pkg/cloud/aws/actuators/cluster/actuator.go

```diff
@@ -122,9 +133,54 @@ func (a *Actuator) Reconcile(cluster *clusterv1.Cluster) error {
 		return errors.Wrapf(err, "failed to get kubeconfig secret for cluster %q", cluster.Name)
 	}
 
+	if cluster.Annotations[v1alpha1.AnnotationControlPlaneReady] != v1alpha1.ValueReady {
+		log.Info("Cluster does not have ready annotation - checking for ready control plane machines")
+
+		machines, err := a.client.Machines(cluster.Namespace).List(actuators.ListOptionsForCluster(cluster.Name))
+		if err != nil {
+			return errors.Wrapf(err, "failed to list machines for cluster %q", cluster.Name)
+		}
+
+		controlPlaneMachines := machine.GetControlPlaneMachines(machines)
+
+		machineReady := false
+		for _, machine := range controlPlaneMachines {
+			if machine.Status.NodeRef != nil {
+				machineReady = true
+				break
+			}
+		}
+
+		if machineReady {
+			log.Info("Setting cluster ready annotation")
+			cluster.Annotations[v1alpha1.AnnotationControlPlaneReady] = v1alpha1.ValueReady
+		} else {
+			log.Info("No control plane machines are ready - requeuing cluster")
+			return &controllerError.RequeueAfterError{RequeueAfter: waitForControlPlaneMachineDuration}
+		}
+	} else {
+		configMapName := actuators.ControlPlaneConfigMapName(cluster)
+		log.Info("Checking for existence of control plane configmap lock", "configmap-name", configMapName)
+
+		_, err := a.coreClient.ConfigMaps(cluster.Namespace).Get(configMapName, metav1.GetOptions{})
+		switch {
+		case apierrors.IsNotFound(err):
+			// It doesn't exist - no-op
+		case err != nil:
+			return errors.Wrapf(err, "Error retrieving control plane configmap lock %q", configMapName)
+		default:
+			if err := a.coreClient.ConfigMaps(cluster.Namespace).Delete(configMapName, nil); err != nil {
+				return errors.Wrapf(err, "Error deleting control plane configmap lock %q", configMapName)
+			}
+		}
+
+	}
+
 	return nil
 }
 
+const waitForControlPlaneMachineDuration = 15 * time.Second
```

Comment:
Nit: move globals at the top of the file

Reply:
I like them close to where they're accessed, but I can move them if you feel strongly.